python/tvm/topi/image/resize.py

```diff
@@ -136,71 +122,96 @@ def get_3d_pixel(data, layout, image_depth, image_height, image_width, n, c, z,
         )
-    return in_x
 
 
-def get_closest_index(in_x, rounding_method, boxes, use_int_div=False):
+def get_closest_index(in_x, rounding_method, boxes):
     """get the closest index to a value based on a certain rounding method"""
-    if use_int_div:
-        closest_x_index = in_x.astype("int32")
-        return closest_x_index
-
-    if rounding_method == "round" or boxes is not None:
-        closest_x_index = te.round(in_x).astype("int32")
-    elif rounding_method == "round_prefer_floor":
-        closest_x_index = te.ceil(in_x - 0.5).astype("int32")
-    elif rounding_method == "round_prefer_ceil":
-        closest_x_index = te.floor(in_x + 0.5).astype("int32")
-    elif rounding_method == "floor":
-        # Add epsilon to floor to prevent gpu rounding errors.
-        epsilon = 1e-5
-        closest_x_index = te.floor(in_x + epsilon).astype("int32")
-    elif rounding_method == "ceil":
-        # Subract epsilon from ceil to prevent gpu rounding errors.
-        epsilon = 1e-5
-        closest_x_index = te.ceil(in_x - epsilon).astype("int32")
+    if isinstance(in_x, Fraction):
+        # Preferred path, if the initial sizes were an integer ratio.
+
+        numerator = in_x.numerator
+        denominator = in_x.denominator
+        if rounding_method in ("round", "round_prefer_floor") or boxes is not None:
+            return (numerator + denominator // 2) // denominator
+        elif rounding_method == "round_prefer_ceil":
+            return (numerator + (denominator + 1) // 2) // denominator
+        elif rounding_method == "floor":
+            return numerator // denominator
+        elif rounding_method == "ceil":
+            return (numerator + denominator - 1) // denominator
+        else:
+            raise ValueError("Uknown rounding method: {}".format(rounding_method))
+
     else:
-        raise ValueError("Uknown rounding method: {}".format(rounding_method))
-    return closest_x_index
+        # Preferred path, using floating-point values
+
+        if rounding_method == "round" or boxes is not None:
+            return te.round(in_x).astype("int32")
+        elif rounding_method == "round_prefer_floor":
+            return te.ceil(in_x - 0.5).astype("int32")
+        elif rounding_method == "round_prefer_ceil":
+            return te.floor(in_x + 0.5).astype("int32")
+        elif rounding_method == "floor":
+            # Add epsilon to floor to prevent gpu rounding errors.
+            epsilon = 1e-5
+            return te.floor(in_x + epsilon).astype("int32")
+        elif rounding_method == "ceil":
+            # Subract epsilon from ceil to prevent gpu rounding errors.
+            epsilon = 1e-5
+            return te.ceil(in_x - epsilon).astype("int32")
+        else:
+            raise ValueError("Uknown rounding method: {}".format(rounding_method))
 
 
 def _lerp(A, B, t):
     """Perform Linear interpolation in 1D"""
-    return A * (1.0 - t) + B * t
+    return (1.0 - t) * A + t * B
```

Comment:
Just for curiosity, why the order matters here?

Reply:
The ordering is necessary for Python's operator overload resolution.  If both arguments for an operator have a `__mul__`, then the left-hand side takes precedence.  Because `t` has already been converted to a `Fraction` (if possible, with fallback to a `PrimExpr` otherwise), I want to use `Fraction.__mul__` instead of `PrimExpr.__mul__`.  There's a similar change that was needed when multiplying the weights and values together, with the weights (`Fraction`) on the lhs and the values (`PrimExpr`) on the rhs.

If the order is reversed (`PrimExpr * Fraction`), then it fails noisily, as `tvm.runtime.convert` cannot handle `Fraction`.  This is intentional, to avoid accidentally converting from `Fraction` to `PrimExpr` prematurely.